src/NextcloudCalendar.php

```diff
@@ -0,0 +1,443 @@
+<?php declare(strict_types=1);
+
+namespace Pdsinterop\Flysystem\Adapter;
+
+use League\Flysystem\AdapterInterface;
+use League\Flysystem\Config;
+use OCP\Files\Folder;
+
+use OCA\DAV\CalDAV\CalDavBackend;
+use OCA\DAV\Connector\LegacyDAVACL;
+use OCA\DAV\CalDAV\CalendarRoot;
+use OCA\DAV\Connector\Sabre\Auth;
+use OCA\DAV\Connector\Sabre\ExceptionLoggerPlugin;
+use OCA\DAV\Connector\Sabre\MaintenancePlugin;
+use OCA\DAV\Connector\Sabre\Principal;
+
+
+/**
+ * Filesystem adapter to convert RDF files to and from a default format
+ */
+class NextcloudCalendar implements AdapterInterface
+{
+    private $defaultAcl;
+    private $userId;
+    private $principalUri;
+
+    final public function __construct($userId, $defaultAcl)
+    {
+        $this->userId = $userId;
+        $this->principalUri = "principals/users/" . $this->userId;
+	$this->defaultAcl = $defaultAcl;
+
+        $authBackend = new Auth(
+            \OC::$server->getSession(),
+            \OC::$server->getUserSession(),
+            \OC::$server->getRequest(),
+            \OC::$server->getTwoFactorAuthManager(),
+            \OC::$server->getBruteForceThrottler(),
+            'principals/'
+        );
+        $principalBackend = new Principal(
+            \OC::$server->getUserManager(),
+            \OC::$server->getGroupManager(),
+            \OC::$server->getShareManager(),
+            \OC::$server->getUserSession(),
+            \OC::$server->getAppManager(),
+            \OC::$server->query(\OCA\DAV\CalDAV\Proxy\ProxyMapper::class),
+            \OC::$server->getConfig(),
+            'principals/'
+        );
+        $db = \OC::$server->getDatabaseConnection();
+        $userManager = \OC::$server->getUserManager();
+        $random = \OC::$server->getSecureRandom();
+        $logger = \OC::$server->getLogger();
+        $dispatcher = \OC::$server->getEventDispatcher();
+
+        $this->calDavBackend = new CalDavBackend($db, $principalBackend, $userManager, \OC::$server->getGroupManager(), $random, $logger, $dispatcher, true);	
```

Comment:
For now this is fine but I think the `CalDavBackend` creation will eventually have to be moved outside of the class.